sys/net/gnrc/routing/rpl/gnrc_rpl.c

```diff
@@ -244,73 +308,38 @@ static void *_event_loop(void *args)
     return NULL;
 }
 
+#ifdef MODULE_GNRC_RPL_P2P
 void _update_lifetime(void)
 {
-    gnrc_rpl_parent_t *parent;
-    gnrc_rpl_instance_t *inst;
-
-    for (uint8_t i = 0; i < GNRC_RPL_PARENTS_NUMOF; ++i) {
-        parent = &gnrc_rpl_parents[i];
-        if (parent->state != 0) {
-            if (parent->lifetime > GNRC_RPL_LIFETIME_UPDATE_STEP) {
-                if (parent->lifetime <= (2 * GNRC_RPL_LIFETIME_UPDATE_STEP)) {
-                    gnrc_rpl_send_DIS(parent->dodag->instance, &parent->addr);
-                }
-                parent->lifetime -= GNRC_RPL_LIFETIME_UPDATE_STEP;
-            }
-            else {
-                gnrc_rpl_dodag_t *dodag = parent->dodag;
-                gnrc_rpl_parent_remove(parent);
-                gnrc_rpl_parent_update(dodag, NULL);
-            }
-        }
-    }
-
-    for (int i = 0; i < GNRC_RPL_INSTANCES_NUMOF; ++i) {
-        inst = &gnrc_rpl_instances[i];
-        if (inst->state != 0) {
-            if ((inst->cleanup > 0) && (inst->dodag.parents == NULL) &&
-                (inst->dodag.my_rank == GNRC_RPL_INFINITE_RANK)) {
-                inst->cleanup -= GNRC_RPL_LIFETIME_UPDATE_STEP;
-                if (inst->cleanup <= 0) {
-                    /* no parents - delete this instance and DODAG */
-                    gnrc_rpl_instance_remove(inst);
-                    continue;
-                }
-            }
-
-            if (inst->dodag.dao_time > GNRC_RPL_LIFETIME_UPDATE_STEP) {
-                inst->dodag.dao_time -= GNRC_RPL_LIFETIME_UPDATE_STEP;
-            }
-            else {
-                _dao_handle_send(&inst->dodag);
-            }
-        }
-    }
-
-#ifdef MODULE_GNRC_RPL_P2P
     gnrc_rpl_p2p_update();
-#endif
 
     xtimer_set_msg(&_lt_timer, _lt_time, &_lt_msg, gnrc_rpl_pid);
 }
+#endif
 
 void gnrc_rpl_delay_dao(gnrc_rpl_dodag_t *dodag)
 {
-    dodag->dao_time = GNRC_RPL_DEFAULT_DAO_DELAY;
+    evtimer_del(&gnrc_rpl_evtimer, (evtimer_event_t *)&dodag->dao_event);
+    ((evtimer_event_t *)&(dodag->dao_event))->offset = GNRC_RPL_DEFAULT_DAO_DELAY + random_uint32_range(500,2000);
+    evtimer_add_msg(&gnrc_rpl_evtimer, &dodag->dao_event, gnrc_rpl_pid);
     dodag->dao_counter = 0;
     dodag->dao_ack_received = false;
 }
 
 void gnrc_rpl_long_delay_dao(gnrc_rpl_dodag_t *dodag)
 {
-    dodag->dao_time = GNRC_RPL_REGULAR_DAO_INTERVAL;
+    evtimer_del(&gnrc_rpl_evtimer, (evtimer_event_t *)&dodag->dao_event);
+    ((evtimer_event_t *)&(dodag->dao_event))->offset = GNRC_RPL_REGULAR_DAO_INTERVAL + random_uint32_range(500,2000);
```

Comment:
I don't *like* these magic numbers to generate the offset jitter, IMHO better do something like:

```
((evtimer_event_t *)&(dodag->dao_event))->offset = random_uint32_range(GNRC_RPL_REGULAR_DAO_DELAY, GNRC_RPL_DAO_DELAY_LONG + GNRC_RPL_DAO_DELAY_JITTER);
```